### PR TITLE
Add isPassword field to secret-type response

### PR DESCRIPTION
**Pull request by colin014**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | partially https://github.com/banzaicloud/banzai-cli/issues/315
| License         | Apache 2.0

### What's in this PR?
Add isPassword field to secret-type response

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] OpenAPI and Postman files updated (if needed)

### CODE REVIEW

File: .gen/pipeline/pipeline/model_secret_type_response_fields.go [hunk 1]
```diff
@@ -16,5 +16,7 @@ type SecretTypeResponseFields struct {
 
 	Required bool `json:"required,omitempty"`
 
+	IsPassword *bool `json:"isPassword,omitempty"`
```
Comment (pregnor) [inline .gen/pipeline/pipeline/model_secret_type_response_fields.go:19]: **Question**: What are we using the nullability for?
If it's unset then it's true, if its set to false then false, if it's set to true then true?

In this case I would propose trying to reverse the field and using it without nullability.
I'm trying to find a non-negated name for the reverse field, but failed so far.
If we can't I'm fine with this, just an idea.

In general I support sensitive default for false value and explicit turn-off.

(If I assumed it wrongly then please clarify it for me and let's talk about it.)

(If we change it, the references need to be updated as well.)

File: .gen/pipeline/api/openapi.yaml [hunk 2]
```diff
@@ -24124,6 +24127,9 @@ components:
           type: string
         required:
           type: boolean
+        isPassword:
+          nullable: true
+          type: boolean
```
Comment (pregnor) [inline .gen/pipeline/api/openapi.yaml:24132]: **Question**: what triggered these changes?
Have you edited the generated file instead of the manual one?

Comment (colin014) [reply .gen/pipeline/api/openapi.yaml:24132]: No, `generate-openapi` task (from Makefile) updated these files in `.gen` directory

File: .gen/pipeline/pipeline/model_secret_type_response_fields.go (hunk 1 above)
Comment (pregnor) [reply .gen/pipeline/pipeline/model_secret_type_response_fields.go:19]: Checked the CLI change and if I understood correctly we are using nil value equals false in behavior so I'm confused why is it nullable.

File: .gen/pipeline/api/openapi.yaml (hunk 2 above)
Comment (pregnor) [reply .gen/pipeline/api/openapi.yaml:24132]: I understood that, but what was the source.

Isn't this change supposed to be coming from apis/pipeline/pipeline.yaml only?

Comment (pregnor) [reply .gen/pipeline/api/openapi.yaml:24132]: I could find it in the `apis/pipeline/pipeline.yaml` now, so it's fixed, I must have missed it on first look I guess.

File: .gen/pipeline/pipeline/model_secret_type_response_fields.go (hunk 1 above)
Comment (pregnor) [reply .gen/pipeline/pipeline/model_secret_type_response_fields.go:19]: We discussed in private my original assumption was right, the CLI code was mistaken.

File: internal/secret/secrettype/secret.go [hunk 3]
```diff
@@ -266,40 +267,40 @@ var DefaultRules = map[string]Meta{
...
 		Fields: []FieldMeta{
-			{Name: OracleUserOCID, Required: true, Description: "Your Oracle user OCID. Find more about, generating public key and fingerprint here: https://banzaicloud.com/docs/pipeline/secrets/providers/oci_auth_credentials/"},
-			{Name: OracleTenancyOCID, Required: true, Description: "Your tenancy OCID"},
-			{Name: OracleAPIKey, Required: true, Description: "Your public key"},
-			{Name: OracleAPIKeyFingerprint, Required: true, Description: "Fingerprint of you public key"},
-			{Name: OracleRegion, Required: true, Description: "Oracle region"},
-			{Name: OracleCompartmentOCID, Required: true, Description: "Your compartment OCID"},
+			{Name: OracleUserOCID, Required: true, IsSafeToDisplay: true, Description: "Your Oracle user OCID. Find more about, generating public key and fingerprint here: https://banzaicloud.com/docs/pipeline/secrets/providers/oci_auth_credentials/"},
+			{Name: OracleTenancyOCID, Required: true, IsSafeToDisplay: true, Description: "Your tenancy OCID"},
+			{Name: OracleAPIKey, Required: true, IsSafeToDisplay: true, Description: "Your public key"},
```
Comment (pregnor) [inline internal/secret/secrettype/secret.go:272]: **Question**: are we sure this is a public key?
API keys are usually private that's why I'm asking, the name and the description are somewhat contradicting each other according to my understanding, but that's just my intuition.

Same [at](https://github.com/banzaicloud/pipeline/pull/3202/files#diff-3aae6f1133ed7ec044650bb3eb2efec4R44) `internal/secret/types/type_oracle.go:44`.

**Review by pregnor — APPROVED** [review]

File: .gen/pipeline/pipeline/model_secret_type_response_fields.go (hunk 1 above)
Comment (pregnor) [reply .gen/pipeline/pipeline/model_secret_type_response_fields.go]: [Fixed](https://github.com/banzaicloud/pipeline/pull/3202/commits/fb1bf55e3d9609c04f7fca8eccb6cd573cbbe7bd).

File: internal/secret/secrettype/secret.go (hunk 3 above)
Comment (colin014) [reply internal/secret/secrettype/secret.go:272]: Yes, this is a public key (https://banzaicloud.com/docs/pipeline/secrets/providers/oci_auth_credentials/#generate-api-token). In Oracle side we need to upload a public key to use this secret

Comment (pregnor) [reply internal/secret/secrettype/secret.go:272]: Cool, it's all good then.

**Pull request merged**